apps/aecore/lib/aecore/multi_node_test_framework/worker.ex

```diff
@@ -0,0 +1,611 @@
+  end
+
+  defp update_oracle_interaction_objects_state(node_name) do
+    GenServer.call(__MODULE__, {:update_oracle_interaction_objects_state, node_name})
+  end
+
+  @spec update_top_block_state(String.t()) :: :ok | String.t()
+  defp update_top_block_state(node_name) do
+    GenServer.call(__MODULE__, {:update_top_block_state, node_name})
+  end
+
+  @spec update_peers_map(String.t()) :: :ok
+  defp update_peers_map(node_name) do
+    GenServer.call(__MODULE__, {:update_peers_map, node_name})
+  end
+
+  # oracles
+  @spec oracle_interaction_objects(String.t()) :: :ok | :unknown_node
+  def oracle_interaction_objects(node_name) do
+    send_command(node_name, "int_object = Chain.oracle_interaction_objects()")
+
+    # converting the keys which are binary to string
+    send_command(
+      node_name,
+      "oracle_int_obj_for_encoding = for {k,v} <- int_object, into: %{}, do: {Base.encode32(k), v}"
+    )
+
+    # converting the keys which are binary to string in the nested map
+    send_command(
+      node_name,
+      "oracles_encoded = Enum.reduce(oracle_int_obj_for_encoding, %{}, fn {k, val}, acc ->
+                                new_oracles = put_in(oracle_int_obj_for_encoding[k].sender_address, Base.encode32(val.sender_address))
+                                put_in(new_oracles[k].oracle_address, Base.encode32(val.oracle_address)) end)"
+    )
+
+    # encoding the map to the json
+    send_command(node_name, "{:ok, json} = Poison.encode(oracles_encoded)")
+
+    # writing it to the file
+    send_command(node_name, "path = System.cwd() <> \"/result.json\"")
+    send_command(node_name, "File.write(path, json)")
+
+    # updating the state of the oracle interaction objects after the one second
+    :timer.sleep(1000)
+    update_oracle_interaction_objects_state(node_name)
+  end
+
+  @spec registered_oracles(String.t()) :: :ok | :unknown_node
+  def registered_oracles(node_name) do
+    send_command(node_name, "registered_oracles = Chain.registered_oracles()")
+
+    # converting the keys which are binary to string
+    send_command(
+      node_name,
+      "oracles_for_encoding = for {k,v} <- registered_oracles, into: %{}, do: {Base.encode32(k), v}"
+    )
+
+    # converting the keys which are binary to string in a nested map
+    send_command(
+      node_name,
+      "oracles_encoded = Enum.reduce(oracles_for_encoding, %{}, fn {k, val}, acc ->
+                                put_in(oracles_for_encoding[k].owner, Base.encode32(val.owner))
+                              end)"
+    )
+
+    # encoding the map to the json
+    send_command(node_name, "{:ok, json} = Poison.encode(oracles_encoded)")
+
+    # writing it to the file
+    send_command(node_name, "path = System.cwd() <> \"/result.json\"")
+    send_command(node_name, "File.write(path, json)")
+
+    # updating the state of the oracle interaction objects after the one second
+    :timer.sleep(1000)
```

Comment:
Sleep is something I really don't like in integration tests as: it makes them slow, they can still fail (and this may be hard to reproduce) because of too short sleep. 

But we should probably leave it as is, because sleep is very hard to get rid of in integration tests.

Reply:
I use sleep here and in other places because the logic of receiving the result from each node is encoding the result and writing it to the json file and after that reading from it, and it needs some time to do this. So if there is a way to change this logic we can remove these timers.